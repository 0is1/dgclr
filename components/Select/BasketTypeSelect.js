// @flow
import React from 'react';
import Select from 'components/Select';

const basketTypeOptions = [
  { value: 'Amexpo', label: 'Amexpo' },
  { value: 'DiscGolfPark-maalikori', label: 'DiscGolfPark-maalikori' },
  { value: 'Knickarp', label: 'Knickarp' },
  { value: 'Latitude64', label: 'Latitude64' },
  { value: 'M-Stone', label: 'M-Stone' },
  { value: 'Muu', label: 'Muu' },
  { value: 'Obsidian Discs', label: 'Obsidian Discs' },
  { value: 'Prodigy', label: 'Prodigy' },
  { value: 'Prodiscus', label: 'Prodiscus' },
];

type Props = { onChange: Function };

const BasketTypeSelect = (props: Props) => {
  const { onChange } = props;
  return (
    <Select
      options={basketTypeOptions}
      onChange={onChange}
      placeholder="Korityyppi"
      filterName="basketType"
      label="Korityyppi:"
    />
  );
};

export default BasketTypeSelect;
