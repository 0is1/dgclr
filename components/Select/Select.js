// @flow
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Label } from 'rebass';
// $FlowFixMe
import SelectInput from 'react-select';
import { setFilter as setFilterFunc } from 'components/SearchContainer/actions';
import { getFilterTypeData } from 'components/SearchContainer/selectors';
// $FlowFixMe
import type { OptionsType, ValueType } from 'react-select/src/types';

type Props = {
  defaultValue: OptionsType,
  filterName: string,
  label: string,
  onChange: Function,
  options: OptionsType,
  placeholder: string,
  setFilter: Function,
};

class Select extends Component<Props> {
  onValueChange = (values: ValueType) => {
    // console.log('values: ', values);
    const { filterName, onChange, setFilter } = this.props;
    const valueData = values ? [].concat(values) : '';
    const value = values && values.value ? values.value : '';
    onChange(value);
    setFilter(filterName, valueData);
  };

  render() {
    const {
      defaultValue, label, options, placeholder,
    } = this.props;
    return (
      <React.Fragment>
        <Label>{label}</Label>
        <SelectInput defaultValue={defaultValue} options={options} onChange={this.onValueChange} placeholder={placeholder} isClearable />
      </React.Fragment>
    );
  }
}

const mapStateToProps = (state, ownProps) => ({
  defaultValue: getFilterTypeData(state, ownProps.filterName),
});
const mapDispatchToProps = dispatch => ({
  setFilter: (filterName, data) => dispatch(setFilterFunc(filterName, data)),
});
export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Select);
