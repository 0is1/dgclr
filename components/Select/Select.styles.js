import styled from 'styled-components';
// import colors from 'components/colors';

const Styles = {
  ZIndexContainer: styled.div`
    position: relative;
    z-index: 4;
  `,
};

export default Styles;
