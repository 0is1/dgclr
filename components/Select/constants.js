// eslint-disable-next-line import/prefer-default-export
export const SELECT_FILTER_NAMES = {
  basketType: { filterName: 'basketType' },
  courseTypes: { filterName: 'courseTypes' },
  rating: { filterName: 'rating' },
  surfaceShapeTypes: { filterName: 'surfaceShapeTypes' },
  teeType: { filterName: 'teeType' },
};
