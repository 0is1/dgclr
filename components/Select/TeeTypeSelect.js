// @flow
import React from 'react';
import Select from 'components/Select';

const teeTypeOptions = [
  { value: 'Tekonurmi', label: 'Tekonurmi' },
  { value: 'TeePad', label: 'TeePad' },
  { value: 'Maapohja', label: 'Maapohja' },
  { value: 'Betonilaatat', label: 'Betonilaatat' },
  { value: 'Kumimatto', label: 'Kumimatto' },
  { value: 'Hiekka/Sora', label: 'Hiekka/Sora' },
  { value: 'Lautalaveri', label: 'Lautalaveri' },
  { value: 'Kivituhka', label: 'Kivituhka' },
  { value: 'Betoni', label: 'Betoni' },
];

type Props = { onChange: Function };

const TeeTypeSelect = (props: Props) => {
  const { onChange } = props;
  return (
    <Select
      options={teeTypeOptions}
      onChange={onChange}
      placeholder="Heittopaikan tyyppi"
      label="Heittopaikan tyyppi:"
      filterName="teeType"
    />
  );
};

export default TeeTypeSelect;
