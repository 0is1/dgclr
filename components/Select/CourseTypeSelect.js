// @flow
import React from 'react';
import Select from 'components/Select';
import { SELECT_FILTER_NAMES } from './constants';
import Styles from './Select.styles';

const { ZIndexContainer } = Styles;

export const courseTypeOptions = [
  {
    value: 'Metsärata',
    label: 'Metsärata',
  },
  {
    value: 'Puistorata',
    label: 'Puistorata',
  },
  {
    value: 'Peltorata',
    label: 'Peltorata',
  },
  {
    value: 'Hiekkakuopparata',
    label: 'Hiekkakuopparata',
  },
  {
    value: 'Hiihtokeskus',
    label: 'Hiihtokeskus',
  },
];

type Props = { onChange: Function, placeholder: string, label: string };

const CourseTypeSelect = (props: Props) => {
  const { onChange, ...other } = props;
  return (
    <ZIndexContainer>
      <Select {...other} options={courseTypeOptions} onChange={onChange} filterName={SELECT_FILTER_NAMES.courseTypes.filterName} />
    </ZIndexContainer>
  );
};

export default CourseTypeSelect;
