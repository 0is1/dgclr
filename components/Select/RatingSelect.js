// @flow
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Label } from 'rebass';
// $FlowFixMe
import Select from 'react-select';
import { setFilter as setFilterFunc } from 'components/AdvancedSearch/actions';
import { getFilterTypeData } from 'components/AdvancedSearch/selectors';
// $FlowFixMe
import type { ValueType } from 'react-select/src/types';
import type { State } from 'lib/types';
import { RATING_OPTIONS } from 'lib/constants';
import { SELECT_FILTER_NAMES } from './constants';
import Styles from './Select.styles';

const { ZIndexContainer } = Styles;

type Props = { onChange: Function };
type MapStateToProps = { defaultValue: any };

type MapDispatchToProps = { setFilter: Function };

type CombinedProps = Props & MapStateToProps & MapDispatchToProps;

class RatingSelect extends Component<CombinedProps> {
  onRatingChange = (values: ValueType = []) => {
    const { onChange, setFilter } = this.props;
    if (values) {
      // console.log('onRatingChange values. ', values);
      const rating = values.map(item => item.value);
      onChange(rating);
      setFilter(SELECT_FILTER_NAMES.rating.filterName, values);
    } else {
      onChange('');
      setFilter(SELECT_FILTER_NAMES.rating.filterName, []);
    }
  };

  render() {
    const { defaultValue } = this.props;
    return (
      <ZIndexContainer zIndex={5}>
        <Label>Radan luokitus:</Label>
        <Select
          defaultValue={defaultValue}
          isMulti
          options={RATING_OPTIONS}
          onChange={this.onRatingChange}
          placeholder="Radan luokitus"
        />
      </ZIndexContainer>
    );
  }
}

const mapStateToProps = (state: State): MapStateToProps => ({
  defaultValue: getFilterTypeData(state, 'rating'),
});
const mapDispatchToProps = (dispatch: Function): MapDispatchToProps => ({
  setFilter: (filterName, data) => dispatch(setFilterFunc(filterName, data)),
});
export default connect<CombinedProps, Props, any, any, any, Function>(
  mapStateToProps,
  mapDispatchToProps,
)(RatingSelect);
