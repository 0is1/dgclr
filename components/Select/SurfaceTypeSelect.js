// @flow
import React from 'react';
import Select from 'components/Select';
import { SELECT_FILTER_NAMES } from './constants';

export const surfaceTypeOptions = [
  {
    value: 'Mäkinen, paljon korkeuseroja',
    label: 'Mäkinen, paljon korkeuseroja',
  },
  {
    value: 'Kumpuileva, jonkin verran korkeuseroja',
    label: 'Kumpuileva, jonkin verran korkeuseroja',
  },
  {
    value: 'Tasainen, vähän korkeuseroja',
    label: 'Tasainen, vähän korkeuseroja',
  },
  { value: 'Tasainen, ei korkeuseroja', label: 'Tasainen, ei korkeuseroja' },
];

type Props = { onChange: Function, placeholder: string, label: string };

const SurfaceTypeSelect = (props: Props) => {
  const { onChange, ...other } = props;
  return (
    <Select {...other} options={surfaceTypeOptions} onChange={onChange} filterName={SELECT_FILTER_NAMES.surfaceShapeTypes.filterName} />
  );
};

export default SurfaceTypeSelect;
