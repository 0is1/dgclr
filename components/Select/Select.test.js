/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import SelectInput from 'react-select';
import { Label } from 'rebass';
import TestStoreProvider from 'jest/TestStoreProvider';
import { DEFAULT_STATE } from 'lib/rootReducer';
import { surfaceTypeOptions } from './SurfaceTypeSelect';
import Select from './Select';

const defaultProps = {
  setFilter: jest.fn(),
  options: surfaceTypeOptions,
  filterName: 'filterName',
  label: 'label',
  onChange: jest.fn(),
  placeholder: 'placeholder',
};

function setup(advancedQueries = {}) {
  const storeData = {
    ...DEFAULT_STATE,
    search: {
      ...DEFAULT_STATE.search,
      advancedQueries,
    },
  };
  const components = (
    <TestStoreProvider storeData={storeData}>
      <Select {...defaultProps} />
    </TestStoreProvider>
  );
  const enzymeWrapper = mount(components);
  return { enzymeWrapper };
}

describe('Select component', () => {
  it('Match snapshot', () => {
    const component = renderer.create(
      <TestStoreProvider storeData={DEFAULT_STATE}>
        <Select {...defaultProps} />
      </TestStoreProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Renders correctly', () => {
    const { enzymeWrapper } = setup();
    expect(enzymeWrapper).toBeDefined();
    expect(enzymeWrapper.find(SelectInput).length).toEqual(1);
    expect(enzymeWrapper.find(SelectInput).prop('defaultValue')).toEqual([]);
    expect(enzymeWrapper.find(SelectInput).prop('options')).toEqual(
      surfaceTypeOptions
    );
    expect(enzymeWrapper.props().children.props.filterName).toEqual(
      'filterName'
    );
    expect(enzymeWrapper.props().children.props.label).toEqual('label');
    expect(enzymeWrapper.find(Label).length).toEqual(1);
    expect(enzymeWrapper.find(Label).text()).toEqual('label');
  });
  it('Renders correctly with defaultValue', () => {
    const advancedQueries = {
      filterName: [surfaceTypeOptions[0]],
    };
    const { enzymeWrapper } = setup(advancedQueries);
    expect(enzymeWrapper).toBeDefined();
    expect(enzymeWrapper.find(SelectInput).length).toEqual(1);
    expect(enzymeWrapper.find(SelectInput).prop('defaultValue')).toEqual([
      surfaceTypeOptions[0],
    ]);
    expect(enzymeWrapper.find(SelectInput).prop('options')).toEqual(
      surfaceTypeOptions
    );
    expect(enzymeWrapper.props().children.props.filterName).toEqual(
      'filterName'
    );
    expect(enzymeWrapper.props().children.props.label).toEqual('label');
    expect(enzymeWrapper.find(Label).length).toEqual(1);
    expect(enzymeWrapper.find(Label).text()).toEqual('label');
  });
});
