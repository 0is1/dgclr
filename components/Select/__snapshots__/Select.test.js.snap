// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component Match snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Select)
    filterName="filterName"
    label="label"
    onChange={[MockFunction]}
    options={
      Array [
        Object {
          "label": "Mäkinen, paljon korkeuseroja",
          "value": "Mäkinen, paljon korkeuseroja",
        },
        Object {
          "label": "Kumpuileva, jonkin verran korkeuseroja",
          "value": "Kumpuileva, jonkin verran korkeuseroja",
        },
        Object {
          "label": "Tasainen, vähän korkeuseroja",
          "value": "Tasainen, vähän korkeuseroja",
        },
        Object {
          "label": "Tasainen, ei korkeuseroja",
          "value": "Tasainen, ei korkeuseroja",
        },
      ]
    }
    placeholder="placeholder"
    setFilter={[MockFunction]}
  />
</Provider>
`;
